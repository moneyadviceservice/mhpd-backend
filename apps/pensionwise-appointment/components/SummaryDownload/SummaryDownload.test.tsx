import React from 'react';
import { render, screen } from '@testing-library/react';
import { SummaryDownload } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

const yesAnswersQuery = {
  t1q2: '1',
  t2q3: '1',
  t3q1: '1',
  age: '3',
};

const notSureAnswersQuery = {
  t1q2: '3',
  t2q3: '3',
  t3q1: '3',
  age: '3',
};

const minimalOptionsQuery = {
  t1q2: '2',
  t2q3: '2',
  t3q1: '2',
  age: '2',
};

describe('SummaryDownload component', () => {
  it('renders yes answers correctly in English', () => {
    render(<SummaryDownload query={{ ...yesAnswersQuery, language: 'en' }} />);
    const container = screen.getByTestId('summary-download-form');
    expect(container).toMatchSnapshot();
  });

  it('renders not sure answers correctly in English', () => {
    render(
      <SummaryDownload query={{ ...notSureAnswersQuery, language: 'en' }} />,
    );
    const container = screen.getByTestId('summary-download-form');
    expect(container).toMatchSnapshot();
  });

  it('renders minimal options correctly in English', () => {
    render(
      <SummaryDownload query={{ ...minimalOptionsQuery, language: 'en' }} />,
    );
    const container = screen.getByTestId('summary-download-form');
    expect(container).toMatchSnapshot();
  });

  it('renders yes answers correctly in Welsh', () => {
    render(
      <SummaryDownload
        testId="summary-download-form-welsh"
        query={{ ...yesAnswersQuery, language: 'cy' }}
      />,
    );
    const container = screen.getByTestId('summary-download-form-welsh');
    expect(container).toMatchSnapshot();
  });

  it('renders not sure answers correctly in Welsh', () => {
    render(
      <SummaryDownload
        testId="summary-download-form-welsh"
        query={{ ...notSureAnswersQuery, language: 'cy' }}
      />,
    );
    const container = screen.getByTestId('summary-download-form-welsh');
    expect(container).toMatchSnapshot();
  });

  it('renders minimal options correctly in Welsh', () => {
    render(
      <SummaryDownload
        testId="summary-download-form-welsh"
        query={{ ...minimalOptionsQuery, language: 'cy' }}
      />,
    );
    const container = screen.getByTestId('summary-download-form-welsh');
    expect(container).toMatchSnapshot();
  });
});
