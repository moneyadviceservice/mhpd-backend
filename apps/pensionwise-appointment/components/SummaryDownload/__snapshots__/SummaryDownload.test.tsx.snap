// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryDownload component renders minimal options correctly in English 1`] = `
<form
  action="https://staging-embedded-journeys.moneyhelper.org.uk/en/summary-document/download"
  data-testid="summary-download-form"
  method="POST"
  target="_blank"
>
  <input
    id="download_appointment_summary_supplementary_benefits"
    name="appointment_summary[supplementary_benefits]"
    type="hidden"
    value="false"
  />
  <input
    id="download_appointment_summary_supplementary_debt"
    name="appointment_summary[supplementary_debt]"
    type="hidden"
    value="false"
  />
  <input
    id="download_appointment_summary_supplementary_ill_health"
    name="appointment_summary[supplementary_ill_health]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_defined_benefit_pensions"
    name="appointment_summary[supplementary_defined_benefit_pensions]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_pension_transfers"
    name="appointment_summary[supplementary_pension_transfers]"
    type="hidden"
    value="false"
  />
  <input
    id="download_appointment_summary_appointment_type"
    name="appointment_summary[appointment_type]"
    type="hidden"
    value="50_54"
  />
  <button
    class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
    type="submit"
  >
    <span />
    Download your summary (PDF)
    <span>
      <svgr-mock
        classname=""
      />
    </span>
  </button>
</form>
`;

exports[`SummaryDownload component renders minimal options correctly in Welsh 1`] = `
<form
  action="https://staging-embedded-journeys.moneyhelper.org.uk/cy/explore-your-options/welsh-summary"
  data-testid="summary-download-form-welsh"
  method="POST"
  target="_blank"
>
  <input
    id="download_appointment_summary_supplementary_benefits"
    name="appointment_summary[supplementary_benefits]"
    type="hidden"
    value="false"
  />
  <input
    id="download_appointment_summary_supplementary_debt"
    name="appointment_summary[supplementary_debt]"
    type="hidden"
    value="false"
  />
  <input
    id="download_appointment_summary_supplementary_ill_health"
    name="appointment_summary[supplementary_ill_health]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_defined_benefit_pensions"
    name="appointment_summary[supplementary_defined_benefit_pensions]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_pension_transfers"
    name="appointment_summary[supplementary_pension_transfers]"
    type="hidden"
    value="false"
  />
  <button
    class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
    type="submit"
  >
    <span />
    Lawrlwytho crynodeb (PDF)
    <span>
      <svgr-mock
        classname=""
      />
    </span>
  </button>
</form>
`;

exports[`SummaryDownload component renders not sure answers correctly in English 1`] = `
<form
  action="https://staging-embedded-journeys.moneyhelper.org.uk/en/summary-document/download"
  data-testid="summary-download-form"
  method="POST"
  target="_blank"
>
  <input
    id="download_appointment_summary_supplementary_benefits"
    name="appointment_summary[supplementary_benefits]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_debt"
    name="appointment_summary[supplementary_debt]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_ill_health"
    name="appointment_summary[supplementary_ill_health]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_defined_benefit_pensions"
    name="appointment_summary[supplementary_defined_benefit_pensions]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_pension_transfers"
    name="appointment_summary[supplementary_pension_transfers]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_appointment_type"
    name="appointment_summary[appointment_type]"
    type="hidden"
    value="standard"
  />
  <button
    class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
    type="submit"
  >
    <span />
    Download your summary (PDF)
    <span>
      <svgr-mock
        classname=""
      />
    </span>
  </button>
</form>
`;

exports[`SummaryDownload component renders not sure answers correctly in Welsh 1`] = `
<form
  action="https://staging-embedded-journeys.moneyhelper.org.uk/cy/explore-your-options/welsh-summary"
  data-testid="summary-download-form-welsh"
  method="POST"
  target="_blank"
>
  <input
    id="download_appointment_summary_supplementary_benefits"
    name="appointment_summary[supplementary_benefits]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_debt"
    name="appointment_summary[supplementary_debt]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_ill_health"
    name="appointment_summary[supplementary_ill_health]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_defined_benefit_pensions"
    name="appointment_summary[supplementary_defined_benefit_pensions]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_pension_transfers"
    name="appointment_summary[supplementary_pension_transfers]"
    type="hidden"
    value="true"
  />
  <button
    class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
    type="submit"
  >
    <span />
    Lawrlwytho crynodeb (PDF)
    <span>
      <svgr-mock
        classname=""
      />
    </span>
  </button>
</form>
`;

exports[`SummaryDownload component renders yes answers correctly in English 1`] = `
<form
  action="https://staging-embedded-journeys.moneyhelper.org.uk/en/summary-document/download"
  data-testid="summary-download-form"
  method="POST"
  target="_blank"
>
  <input
    id="download_appointment_summary_supplementary_benefits"
    name="appointment_summary[supplementary_benefits]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_debt"
    name="appointment_summary[supplementary_debt]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_ill_health"
    name="appointment_summary[supplementary_ill_health]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_defined_benefit_pensions"
    name="appointment_summary[supplementary_defined_benefit_pensions]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_pension_transfers"
    name="appointment_summary[supplementary_pension_transfers]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_appointment_type"
    name="appointment_summary[appointment_type]"
    type="hidden"
    value="standard"
  />
  <button
    class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
    type="submit"
  >
    <span />
    Download your summary (PDF)
    <span>
      <svgr-mock
        classname=""
      />
    </span>
  </button>
</form>
`;

exports[`SummaryDownload component renders yes answers correctly in Welsh 1`] = `
<form
  action="https://staging-embedded-journeys.moneyhelper.org.uk/cy/explore-your-options/welsh-summary"
  data-testid="summary-download-form-welsh"
  method="POST"
  target="_blank"
>
  <input
    id="download_appointment_summary_supplementary_benefits"
    name="appointment_summary[supplementary_benefits]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_debt"
    name="appointment_summary[supplementary_debt]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_ill_health"
    name="appointment_summary[supplementary_ill_health]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_defined_benefit_pensions"
    name="appointment_summary[supplementary_defined_benefit_pensions]"
    type="hidden"
    value="true"
  />
  <input
    id="download_appointment_summary_supplementary_pension_transfers"
    name="appointment_summary[supplementary_pension_transfers]"
    type="hidden"
    value="true"
  />
  <button
    class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-secondary-button"
    type="submit"
  >
    <span />
    Lawrlwytho crynodeb (PDF)
    <span>
      <svgr-mock
        classname=""
      />
    </span>
  </button>
</form>
`;
